### ISSUE TITLE
Resolve inconsistency between `dplyr` and `galah` implementations of `collapse()` and `compute()`

### ISSUE THREAD

**Issue by mjwestgate** [AtlasOfLivingAustralia/galah-R#217]
As of version 2.0, you can use `collapse()` to convert a `data_request` into a `query_set`. A `query_set` often contains >1 `query`, with the final `query` being the one requested by the user, and earlier `query`s being needed to check that it is valid. Finally, calling `compute()` evaluates the `query_set` into a single, valid URL:
```
x <- galah_call() |>
    filter(year == 2010) |>
    count() |>
    collapse()
x
Object of class `query_set` containing 3 queries:
• metadata/fields data: galah:::check_internal_cache()$fields
• metadata/assertions data: galah:::check_internal_cache()$assertions
• data/occurrences-count url: https://biocache-ws.ala.org.au/ws/occurrences/sea...

compute(x)$url
[1] "https://biocache-ws.ala.org.au/ws/occurrences/search?fq=%28year%3A%222010%22%29&disableAllQualityFilters=true&pageSize=0"
```

For occurrence queries, however, `compute()` is actually doing three things:
- evaluating (some) API calls to check the request is valid
- building a valid url
- sending that API call to the selected atlas

This is not ideal, because it means that for occurrence queries, it is impossible to interrogate the URL before sending, or to see it once it has been sent. Fundamentally, this problem occurs because of a conflict between the principle that `collapse()` shouldn't ping an API, and the need in `galah` to run checks before building a URL. I see two possible solutions:

- remove the limitation on `collapse()` to not hit any APIs; then `collapse()` would always return a valid `query` (or error if needed), but would also run checks
- add a function _between_ `collapse()` and `compute()` (perhaps `precompute()`?) to build a single URL, but not push the request to the selected atlas

My instinct is that the former is neater, though it does change some behaviour from version 2.0. For context, this is what `dplyr` says about these functions:
```
compute() stores results in a remote temporary table. collect() retrieves data 
into a local tibble. collapse() is slightly different: it doesn't force computation, 
but instead forces generation of the SQL query. This is sometimes needed to 
work around bugs in dplyr's SQL generation.
```

**Comment by mjwestgate** [comment]
Decision on 17-01-2024:
- `collapse()` will now return a `query` object, which by default has slots `type` and `url`
  - i.e. previous behaviour of `compute()`
- `collapse()` also gains a new `.expand` (?) argument
  - when `TRUE`, this appends the `query_set` to an additional slot (possible called `calls`?) to support debugging
- `compute()` has reduced functionality under this model
  -  mainly acts as a pass-trough function between `collapse()` and `compute()`
  - `compute()` is still critical for `type = "occurrences"`, where it submits the query (but doesn't wait for a response)

**Comment by mjwestgate** [comment]
Implemented in v 2.0.1
